tools/dds/dds-server/lrs-device-controller.cpp

```diff
@@ -446,7 +467,49 @@ void lrs_device_controller::start_streaming( const json & msg )
     _dds_device_server->start_streaming( realdds_streams_to_start );
     _sensors[sensor_index].open( rs_profiles_to_open );
     _sensors[sensor_index].start( [&]( rs2::frame f ) {
-        _dds_device_server->publish_image( f.get_profile().stream_name(), static_cast< const uint8_t * >( f.get_data() ), f.get_data_size() );
+        const auto & stream_name = f.get_profile().stream_name();
+        auto & streams = _dds_device_server->get_streams();
+
+        auto it = streams.find( stream_name );
+        if( it == streams.end() )
+            throw std::runtime_error( "stream '" +stream_name + "' does not exist" );
+        auto & stream = it->second;
+        if( !stream->is_streaming() )
+            throw std::runtime_error( "stream '" +stream_name + "' is not streaming" );
+
+        stream->publish( static_cast< const uint8_t * >( f.get_data() ), f.get_data_size(), f.get_frame_number() );
+
+        if( _md_enabled )
+        {
+            const auto & metadata_stream_name = f.get_profile().stream_name() + " metadata";
+
+            auto it = streams.find( metadata_stream_name );
+            if( it == streams.end() )
+                throw std::runtime_error( "stream '" + metadata_stream_name + "' does not exist" );
+            auto metadata_stream = std::dynamic_pointer_cast< realdds::dds_metadata_stream_server >( it->second );
+            if( !metadata_stream )
+                throw std::runtime_error( "stream '" + stream_name + "' does not have a matching metadata stream" );
+
+            json metadata = json( {
+                // Special data that is always expected by realdds
+                { "frame_id", std::to_string( f.get_frame_number() ) },
+                { "timestamp", f.get_timestamp() },
+                { "timestamp_domain", f.get_frame_timestamp_domain() }
+                } );
+            if( f.is< rs2::depth_frame >() )
+                metadata["depth_units"] = f.as< rs2::depth_frame >().get_units();
```

Comment:
Is `depth_units` some special requirement? Otherwise I'd use `depth-units`.

Reply:
Changed